crates/interpreter/src/valid.rs

```diff
@@ -742,17 +778,21 @@ impl<'a, 'm> Expr<'a, 'm> {
         self.for_each(expected, |x, y| check(x.matches(y)))
     }
 
-    fn push_label(&mut self, type_: FuncType<'m>, kind: LabelKind) -> CheckResult {
+    fn push_label(&mut self, type_: FuncType<'m>, kind: LabelKind<'m>) -> CheckResult {
         self.pops(type_.params)?;
         let stack = type_.params.iter().cloned().map(OpdType::from).collect();
-        let label = Label { type_, kind, polymorphic: false, stack };
+        let label = Label { type_, kind, polymorphic: false, stack, branches: vec![] };
         self.labels.push(label);
         Ok(())
     }
 
     fn end_label(&mut self) -> CheckResult {
+        for branch in core::mem::take(&mut self.label().branches) {
+            self.stitch(&branch);
+        }
+
```

Comment:
Just wondering your thought behind this. I tend to use empty lines to create "paragraphs" as independent units/steps within a function.

Reply:
Yes, I also do that, but only if the function is long and shouldn't be split in smaller functions. This function is quite short, adding vertical spacing just reduces readability instead of increasing it. Said otherwise, vertical space is useful if the function doesn't fit on the screen, by adding anchors in the form of empty lines (ideally with a comment which would be the name of the helper function if the function were to be split).